ipatests/test_ipaserver/test_install/test_installer.py

```diff
@@ -0,0 +1,185 @@
+#
+# Copyright (C) 2018  FreeIPA Contributors.  See COPYING for license
+#
+from __future__ import absolute_import
+
+import six
+
+if six.PY2:
+    from abc import ABCMeta
+
+    class ABC(object):
+        __metaclass__ = ABCMeta
+else:
+    from abc import ABC  # pylint: disable=no-name-in-module
+
+from abc import abstractmethod
+from collections import namedtuple
+import itertools
+import tempfile
+
+from ipatests.util import assert_equal
+from ipaserver.install.ipa_replica_install import ReplicaInstall
+
+Keyval = namedtuple('Keyval', ['option', 'value'])
+
+
+class InstallerTestBase(ABC):
+    OPTS_DICT = {}
+
+    # don't allow creating classes with tested_cls unspecified
+    @property
+    @abstractmethod
+    def tested_cls(self):
+        return None
+
+    def setup_class(self):
+        """Initializes the tested class so that it can be used later on
+        """
+        self.tested_cls.make_parser()
+        assert \
+            getattr(self.tested_cls, 'option_parser', False), \
+            ("Unable to generate option parser for {}"
+             .format(self.tested_cls.__name__))
+
+        self._populate_opts_dict()
+
+    @classmethod
+    def _populate_opts_dict(cls):
+        """Populate the class-owned OPTS_DICT with available options
+        """
+        if not getattr(cls.tested_cls, 'option_parser', False):
+            raise RuntimeError("You need to create the parser of the tested "
+                               "class first.")
+
+        # add all options from the option groups
+        # pylint: disable=no-member
+        for opt_group in cls.tested_cls.option_parser.option_groups:
+            for opt in opt_group.option_list:
+                cls.OPTS_DICT[opt.dest] = opt._short_opts + opt._long_opts
+        # add options outside groups
```

Comment:
We probably cannot get this part to parse that way so we can have only one loop, right? Just asking...

Reply:
I don't think we can do that in a way that'd be more effective.